src/sql/workbench/contrib/notebook/test/browser/notebookActions.test.ts

```diff
@@ -0,0 +1,145 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as TypeMoq from 'typemoq';
+import * as assert from 'assert';
+
+import { AddCellAction, ClearAllOutputsAction, CollapseCellsAction, TrustedAction, RunAllCellsAction, NewNotebookAction } from 'sql/workbench/contrib/notebook/browser/notebookActions';
+import { CellType } from 'sql/workbench/contrib/notebook/common/models/contracts';
+import { INotebookEditor } from 'sql/workbench/services/notebook/browser/notebookService';
+import { NotebookComponentStub, CommandServiceStub } from 'sql/workbench/contrib/notebook/test/browser/common';
+import { ICellModel, INotebookModel } from 'sql/workbench/contrib/notebook/browser/models/modelInterfaces';
+import { INotificationService } from 'vs/platform/notification/common/notification';
+import { TestNotificationService } from 'vs/platform/notification/test/common/testNotificationService';
+import { ICommandService } from 'vs/platform/commands/common/commands';
+
+suite('Notebook Actions', function (): void {
+	test('Add Cell Action', async function (): Promise<void> {
+		let testCellType: CellType = 'code';
+		let actualCellType: CellType;
+
+		let mockNotebookComponent = TypeMoq.Mock.ofType<INotebookEditor>(NotebookComponentStub);
+		mockNotebookComponent.setup(c => c.addCell(TypeMoq.It.isAny())).returns(cellType => {
+			actualCellType = cellType;
+		});
+
+		let action = new AddCellAction('TestId', 'TestLabel', 'TestClass');
+		action.cellType = testCellType;
+		let result = await action.run(mockNotebookComponent.object);
+
+		assert.ok(result, 'Add Cell Action should succeed');
+		assert.strictEqual(actualCellType, testCellType);
+
+		mockNotebookComponent.reset();
```

Comment:
It would be good to say what is being tested in these tests at each stage. It's not always really clear what it is and so having it split up by adding comments for each new sub-test would help with that.